### ISSUE TITLE
shairport-sync cygwin crash

### ISSUE THREAD

**Issue by noelhibbard** [mikebrady/shairport-sync#396]
I just compiled the latest stable for cygwin and I am noticing a crash that I can reproduce 100% of the time.

Steps to reproduce:
1: Start shairport-sync.exe with no parameters.
2: From the phone, connect to shairport-sync and start playback.
3: From the phone, switch back to local playback.
4: From the phone, switch back to shairplay-sync.

When performing step 4 you get a message on the phone that says **Unable to connect to "C8"** (C8 is the name of my computer) and you will see shairport-sync.exe stops running with no error message. I upped the log verbose to 3 and still don't see any error messages.

I am running shairport-sync on Win10x64 and the phone is an iPhone 7 running iOS 10.0.2.

**Comment by mikebrady** [comment]
Thanks. I'll have a look, though I'm not there yet with an iPhone 7 :)

**Comment by mikebrady** [comment]
Hi there. I've been cleaning up code that I suspect, but can't prove, is possibly causing problems. It's now on the `development` branch. I'd be obliged if you tried it out, to see it it's any better.

**Comment by noelhibbard** [comment]
I just cloned the dev branch and compiled it but as soon as I start streaming to it I get this message:
"unexpected sample rate!"

The process then closes.

I rolled back to the last version I had compiled and it works fine.

**Comment by mikebrady** [comment]
Thanks. I fixed this in the master branch yesterday. I'll fix it in the master branch later today. Sorry about that.

**Comment by mikebrady** [comment]
Just fixed it in the `development` version.

**Comment by noelhibbard** [comment]
Cool.. let me try again.

**Comment by noelhibbard** [comment]
Okay... fixed the sample rate problem but it still crashes using the same steps in my original post. Here is the verbose output if it helps:
https://gist.github.com/noelhibbard/52f727f66cb13680950178d6c603187b

**Comment by noelhibbard** [comment]
FYI, I just checked out the latest master branch and it still has this same crash.

**Comment by mikebrady** [comment]
Thanks. I'll keep looking.

**Comment by mikebrady** [comment]
Hi again. Is it the case that Shairport Sync is crashing or just becoming non-available?

**Comment by noelhibbard** [comment]
Crashing

**Comment by mikebrady** [comment]
Yikes! Thanks.

**Comment by mikebrady** [comment]
Hi Noel. I'm pretty sure Shairport Sync runs in Cygwin now, so let me close this issue. Please open a new one if necessary.

**Comment by noelhibbard** [comment]
I can confirm it no longer crashes in Cygwin.

**Comment by mikebrady** [comment]
Thanks Noel!
